src/flowMax.coffee

```diff
@@ -1,31 +1,48 @@
 import {isAddPropTypes} from './addPropTypes'
 import {isRenderNothing} from './renderNothing'
 import {isReturns} from './returns'
+import {isAddWrapper} from './addWrapper'
+
+getArgumentsPropertyName = '__ad-hok-flowMax-getArguments'
+
+isFlowMax = (func) ->
+  func[getArgumentsPropertyName]
 
 flowMax = (...funcs) ->
-  flowLength = funcs?.length ? 0
-  if flowLength
```

Comment:
In order to support inlining nested `flowMax()`'s, had to adjust `flowMax()` to wait until it actually gets invoked to do any processing of magic helpers (since eg an `addPropTypes()` in a nested `flowMax()` would optimistically apply itself to just the things after it in the nested `flowMax()` rather than getting applied to everything after it in the nested `flowMax()` and after the nested `flowMax()` in the outer `flowMax()`)